internal/db/subscription.go

```diff
@@ -0,0 +1,67 @@
+package db
+
+import (
+	"context"
+	"github.com/ProtonMail/gluon/imap"
+	"github.com/ProtonMail/gluon/internal/db/ent"
+	"github.com/ProtonMail/gluon/internal/db/ent/subscription"
+	"github.com/ProtonMail/gluon/internal/ids"
+)
+
+func AddSubscription(ctx context.Context, tx *ent.Tx, mboxName string, mboxID ids.MailboxIDPair) error {
+	count, err := tx.Subscription.Update().Where(subscription.NameEqualFold(mboxName)).SetMailboxID(mboxID.InternalID).SetRemoteID(mboxID.RemoteID).Save(ctx)
```

Comment:
I'm happy with how you did this, no need to change it, but just fyi we might be able to use upsert here (https://entgo.io/docs/feature-flags#upsert)

Reply:
Noted!